docs: Update and refresh homepage of superset.apache.org

ceohockey60 opened the pull request:
Update and refresh index with additional links to community resources, new links to `Databases' section directly to parts of the tutorial that talk about that database,  connect, align Superset topline features with same language as in GitHub

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

ceohockey60 committed:
Update and refresh Index

Update and refresh index with additional links to community resources, new links to `Databases' section directly to parts of the tutorial that talk about that database,  connect, align Superset topline features with same language as in GitHub

ceohockey60 committed:
Added link to Superset users

etr2460 commented on docs/index.rst:
```diff
@@ -79,54 +83,50 @@ Overview
 The following RDBMS are currently supported:
 
-- `Amazon Athena <https://aws.amazon.com/athena/>`_
+- `Amazon Athena <https://superset.incubator.apache.org/installation.html#aws-athena>`_
```
if this is linking to a page in the docs, do we need the full url? I'd guess `/installation.html#aws-athena` would be sufficient

ceohockey60 replied:
I just tried and it doesn't work, probably because it's .rst so the whole site is generated differently and doesn't work the same way? (I'm no .rst expert...)

etr2460 replied:
I validated that the following change works for me when building the html and running locally, I think this should be safe to do:
```diff
diff --git a/docs/index.rst b/docs/index.rst
index c1af917c1..42d9c28a2 100644
--- a/docs/index.rst
+++ b/docs/index.rst
@@ -99,7 +99,7 @@ Databases
 
 The following RDBMS are currently supported:
 
-- `Amazon Athena <https://superset.incubator.apache.org/installation.html#aws-athena>`_
+- `Amazon Athena </installation.html#aws-athena>`_
 - `Amazon Redshift <https://aws.amazon.com/redshift/>`_
 - `Apache Drill <https://superset.incubator.apache.org/installation.html#apache-drill>`_
 - `Apache Druid <https://superset.incubator.apache.org/installation.html#druid>`_
```

ceohockey60 replied:
Awesome, thanks for validating. Updated all the links accordingly.

ceohockey60 committed:
Update with relative links

etr2460 commented on docs/index.rst:
```diff
@@ -79,54 +83,50 @@ Overview
-- `Vertica <https://www.vertica.com/>`_
-- `Hana <https://www.sap.com/products/hana.html>`_
+- `Amazon Athena </installation.html#aws-athena>`_
+- `Amazon Redshift </installation.html#database-dependencies>`_
```
What's the value in linking to the generic page here instead of the more specific db website?

ceohockey60 replied:
The "database dependencies" table has some information re installation for all the db options, but some db options have their own dedicated anchor tag (I'm not sure why the docs are written this way...). For the ones that have their own tag, it's link to there, but ones that don't, the link points to the "database dependencies" table for community users, which is more helpful than the homepage of the db itself with useful info as it relates to Superset.

ceohockey60 committed:
Add link to docs repo

Add link to docs repo on the homepage to encourage easy access for community contribution.

ceohockey60 committed:
Update index.rst
